### PR TITLE
Upgrade to serde v1.0.0

### PR DESCRIPTION
**Pull request by kardeiz**
This is continuing the work of @TedDriggs's PR (https://github.com/nox/serde_urlencoded/pull/19) to upgrade to Serde 1.0.

I've added wrappers for `Parse` and the `Parse` iterator's key/value `Cow<str>` items, so that they can be properly deserialized according to the output type.

I hope this works (I'm a Git novice); I just pulled @TedDriggs's latest rev and worked from there.

### CODE REVIEW

File: src/ser/mod.rs [hunk 1]
```diff
@@ -98,17 +98,13 @@ pub struct SeqSerializer<'output, Target: 'output + UrlEncodedTarget> {
...
+    urlencoder: &'output mut UrlEncodedSerializer<Target>,
 }
 
 /// Tuple struct serializer.
-///
-/// Never instantiated, tuple structs are not supported.
-pub struct TupleStructSerializer<'output, T: 'output + UrlEncodedTarget> {
-    inner: ser::Impossible<&'output mut UrlEncodedSerializer<T>, Error>,
+pub struct TupleStructSerializer<'output, Target: 'output + UrlEncodedTarget> {
+    urlencoder: &'output mut UrlEncodedSerializer<Target>,
```
Comment (nox) [inline src/ser/mod.rs]: Why did you change that?

Comment (kardeiz) [inline src/ser/mod.rs]: Not sure why this was changed. @TedDriggs?

**Comment by theduke** [comment]
Anything I can do to help? 

Really need a new release.

**Comment by nox** [comment]
I'm on it today.

File: src/ser/mod.rs (hunk 1 above)
Comment (nox) [inline src/ser/mod.rs]: Seems like this is to somehow support the now gone serialize_seq_fixed_size... I think for now I'm going to make a release that doesn't include that.

**Comment by nox** [comment]
Superseded by #23.

Thanks for your contribution.

File: src/ser/mod.rs (hunk 1 above)
Comment (TedDriggs) [inline src/ser/mod.rs]: @nox you got it correct; I was trying to replace the fixed-size-sequence removal.
